### PR TITLE
Fixed typos in abi-spec documentation

### PR DESCRIPTION
**Pull request by nisdas**

### CODE REVIEW

**Comment by chriseth** [comment]
@nisdas I took the liberty to explain the components of X a little better. Also, I think that `k >= 0` is intentional to allow zero-length tuples.

**Comment by chriseth** [comment]
@axic, please review

File: docs/abi-spec.rst [hunk 1]
```diff
@@ -97,7 +97,7 @@ We distinguish static and dynamic types. Static types are encoded in-place and d
 * ``bytes``
 * ``string``
 * ``T[]`` for any ``T``
-* ``T[k]`` for any dynamic ``T`` and any ``k > 0``
+* ``T[k]`` for any dynamic ``T`` and any ``k >= 0``
```
Comment (axic) [inline docs/abi-spec.rst]: Why is `T[0]` valid in the first place?

Comment (chriseth) [reply docs/abi-spec.rst]: Why not? It reduces the amount of exceptional cases.

**Comment by chriseth** [comment]
@axic to merge or not to merge? I'm fine with disallowing `k==0` if you insist.

**Comment by axic** [comment]
> @axic to merge or not to merge? I'm fine with disallowing k==0 if you insist.

I don't mind. `T[0]` feels useless though.

**Comment by chriseth** [comment]
Will rebase.

**Comment by axic** [comment]
So we went for allowing a useless construct of empty arrays? :)

**Comment by axic** [comment]
Probably it made sense merging this as is because it matches the code, but I think it should be discussed for both tuples and fixed-length arrays. Created an issue for that to receive more input.

**Comment by chriseth** [comment]
Actually I realized that we need empty statically-sized arrays: Otherwise, it is undefined how to encode a dynamic array of length zero.

**Pull request merged**